packages/react/__tests__/exports.ts

```diff
@@ -1437,3 +1441,84 @@ describe('@aws-amplify/ui-react/legacy', () => {
     });
   });
 });
+
+const getCatalogJSON = (): PrimitiveCatalog => {
+  try {
+    const rawJSON = fs
+      .readFileSync(path.join(__dirname, '../dist/primitives.json'))
+      .toString();
+
+    return JSON.parse(rawJSON) as PrimitiveCatalog;
+  } catch (err) {
+    console.error('Error reading primitives catalog JSON file:', err);
+  }
+
+  return {};
+};
+
+describe('primitive catalog', () => {
+  const catalog = getCatalogJSON();
+
+  it.each(Object.entries(catalog))(
+    'should contain properties for %s primitive',
+    (name, primitive) => {
+      expect(Object.keys(primitive.properties).length).toBeGreaterThan(0);
+    }
+  );
+
+  it('should match primitives list snapshot', () => {
```

Comment:
Since we've been updating snapshots freely, what do you think about being explicit about the contract?

```suggestion
  it('will have regressions if any of these primitives are missing', () => {
```

🙃